### ISSUE TITLE
Is it possible to assign a function to both up and down on a keypress?

### ISSUE THREAD

**Issue by codemicro** [Gum-Joe/2Keys#21]
I am trying to assign different functions to my keypress that are triggered when I press the key and when I release it as part of a push-to-talk function. 

My `config.yml` shows
```
$NUM_.$:
   type: down
   func: pressPTT
$NUM_.$:
   type: up
   func: releasePTT
```
This is the error that is returned in the logs.
```
{ YAMLSemanticError: Map keys must be unique; "$NUM_.$" is repeated
    at Object.parseMap [as resolve] (c:\2keys\server\node_modules\yaml\dist\schema\parseMap.js:74:27)
    at Schema.resolveNode (c:\2keys\server\node_modules\yaml\dist\schema\index.js:178:27)
    at Schema.resolveNodeWithFallback (c:\2keys\server\node_modules\yaml\dist\schema\index.js:200:22)
    at Document.resolveNode (c:\2keys\server\node_modules\yaml\dist\Document.js:491:22)
    at resolveBlockMapItems (c:\2keys\server\node_modules\yaml\dist\schema\parseMap.js:196:51)
    at Object.parseMap [as resolve] (c:\2keys\server\node_modules\yaml\dist\schema\parseMap.js:38:79)
    at Schema.resolveNode (c:\2keys\server\node_modules\yaml\dist\schema\index.js:178:27)
    at Schema.resolveNodeWithFallback (c:\2keys\server\node_modules\yaml\dist\schema\index.js:200:22)
    at Document.resolveNode (c:\2keys\server\node_modules\yaml\dist\Document.js:491:22)
    at resolveBlockMapItems (c:\2keys\server\node_modules\yaml\dist\schema\parseMap.js:196:51)
  name: 'YAMLSemanticError',
  message: 'Map keys must be unique; "$NUM_.$" is repeated',
  source:
   Collection {
     context:
      ParseContext {
        parseNode: [Function],
        atLineStart: true,
        inCollection: false,
        inFlow: false,
        indent: 4,
        lineStart: 147,
        parent: [CollectionItem],
        root: [Document],
        src:
         'name: servKeyb\nkeyboards:\n  keyboard_1:\n    path: /dev/input/by-id/usb-04d9_1203-event-kbd\n    dir: ./Keyboard_1\n    root: index.ahk\n    hotkeys: \n      $NUM_ENTER$: helloWorld\n      $NUM_.$:\n        type: down\n        func: pressPTT\n      $NUM_.$:\n        type: up\n        func: releasePTT\naddresses:\n  detector: 192.168.5.253\n  server:\n    ipv4: 192.168.5.252\n    port: 9090\nperms:\n  ssh: true\n' },
     error: null,
     range: Range { start: 153, end: 291 },
     valueRange: Range { start: 153, end: 290 },
     props: [],
     type: 'MAP',
     value: null,
     items:
      [ [PlainValue],
        [CollectionItem],
        [PlainValue],
        [CollectionItem],
        [PlainValue],
        [CollectionItem] ],
     resolved: YAMLMap { items: [Array], range: [Array], type: 'MAP' } } }
```
Is there a way to do this?

**Comment by Gum-Joe** [comment]
Not quite yet I am afraid, however, should be easy to implement. The error your getting is because the YAML parser I'm using won't allow the same property to be used twice (in this case, you can only assign $NUM_.$ once).  Example syntax will probably be (therefore) :
```yml
NUM0:
   type: mutli
   up: SomeFunc()
   down: SomeOtherFunc()
```

Also, if you wouldn't please mind me asking, what is your use case for 2Keys?

**Comment by Gum-Joe** [comment]
(BTW please keep this issue open until this has been added. Next time I'm on the PC I'll add it to the feature tracker)

**Comment by codemicro** [comment]
My use case for 2Keys is to create a macropad from a number pad for commonly used functions for virtual air traffic controlling.

The actual functions themself are simple. 
```
pressPTT(){
	Send "{/ Down}"
}

releasePTT(){
	Send "{/ Up}"
}
```

**Comment by Gum-Joe** [comment]
Nice & thanks! (my personal use case is editing macros) 
And also thanks for using 2Keys!

**Comment by codemicro** [comment]
Do you know how it may take for the feature to be implemented?

**Comment by Gum-Joe** [comment]
Sorry for the late reply, I was busy yesterday, but I'm implementing it now.

**Comment by Gum-Joe** [comment]
Please see #22

**Comment by Gum-Joe** [comment]
BTW I've changed the config it's using to:
```yml
SomeKey:
   type: multi # Type key should be able to be safely omitted
   func:
      up: SomeFunc
      down: SomeOtherFunc
```

**Comment by Gum-Joe** [comment]
#22 Merged

**Comment by Gum-Joe** [comment]
Published with v0.4.0.
